src/segger/data/io.py

```diff
@@ -185,7 +186,7 @@ def load_transcripts(
         transcripts_df = self.filter_transcripts(transcripts_df)
 
         # Handle additional embeddings if provided
-        if not self.embedding_df.empty:
+        if self.embedding_df is not None and not self.embedding_df.empty:
```

Comment:
This fixes a bug that was causing an exception when the user was not passing the embedding data.